radis/lbl/broadening.py

```diff
@@ -372,6 +379,77 @@ def pressure_broadening_HWHM(
+    """(This function is based on exojax.spec.atomll.gamma_vald3)
+    
+    HWHM of Lorentzian (cm-1) caluculated as gamma/(4*pi*c) [cm-1] for lines
+    with the van der Waals gamma in the line list (VALD or Kurucz), otherwise
+    estimated according to the Unsoeld (1955)
+
+    Args:
+      T: temperature (K)
+      P: Pressure #1 bar = 1e6 dyn/cm2
+      nu_lines:  transition waveNUMBER in [cm-1] (NOT frequency in [s-1])
+      elower: excitation potential (lower level) [cm-1]
+      ionE: ionization potential [eV]
+      gamRad: log of gamma of radiation damping (s-1) (https://www.astro.uu.se/valdwiki/Vald3Format)
+      gamSta: log of gamma of Stark damping ((s * Nelec)^-1)
+      vdWdamp:  log of (van der Waals damping constant / neutral hydrogen number) (s-1)
+      enh_damp: empirical "enhancement factor" for classical Unsoeld's damping constant cf.) This coefficient (enh_damp) depends on  each species in some codes such as Turbospectrum. #tako210917
+
+    Returns:
+      gammma_rad, gamma_stark, gamma_vdw - Radiation, Stark, and Van der Waals HWHM [cm^-1]
+
+    * Reference of van der Waals damping constant (pressure/collision gamma):
+    *   Unsöld1955: https://ui.adsabs.harvard.edu/abs/1955psmb.book.....U
+    *   Kurucz+1981: https://ui.adsabs.harvard.edu/abs/1981SAOSR.391.....K
+    *   Barklem+1998: https://ui.adsabs.harvard.edu/abs/1998MNRAS.300..863B
+    *   Barklem+2000: https://ui.adsabs.harvard.edu/abs/2000A&AS..142..467B
+    *   Gray+2005: https://ui.adsabs.harvard.edu/abs/2005oasp.book.....G
+    """
+    # based on: https://github.com/HajimeKawahara/exojax/blob/78466cef0170ee1a2768b6a6f7b7c911d715c1bd/src/exojax/spec/atomll.py#L40
+    gamRad = np.where(gamRad == 0., -99, gamRad)
+    gamSta = np.where(gamSta == 0., -99, gamSta)
+    chi_lam = nu_lines/eV2wn  # [cm-1] -> [eV]
+    chi = elower/eV2wn  # [cm-1] -> [eV]
+
+    C6 = 0.3e-30 * ((1/(ionE-chi-chi_lam)**2) - (1/(ionE-chi)**2))
+    C6 = np.abs(C6)
+    
+    weighted_coeff = {}
+    #assuming diluent is dictionary
+    for key in diluent:
+        if key != 'e-' and key in atomic_broadening_coeff:
+            weighted_coeff[key] = diluent[key]*atomic_broadening_coeff[key]
+    gamma6 = 0
+    for key in weighted_coeff:
+        gamma6 += 1e20 * C6**0.4 * P*1e6*weighted_coeff[key] / T**0.7
+    gamma6 *= enh_damp
+    gamma_case1 = gamma6 / (4*np.pi*c_CGS)
+    gamma_case1 = np.nan_to_num(gamma_case1)
+
+    Texp = 0.38  # Barklem+2000
+    gamma6 = 0
+    for key in weighted_coeff:
+        gamma6 += 10**vdWdamp * (T/10000.)**Texp * P*1e6*weighted_coeff[key] / (k_b_CGS*T)
+    gamma_case2 =  gamma6 / (4*np.pi*c_CGS)
+
+    gamma_vdw = np.where(vdWdamp >= 0., gamma_case1, gamma_case2)
+    # df['shft'] = (1.0/3.0)*2*gamma #Konjević et al. 2012 §4.1.3.2
+    #print(gamma_vdw)
+    
+    if 'e-' in diluent:
+        gamma_stark = (10**gamSta) * P*1e6*diluent['e-'] / (k_b_CGS*T) / (4*np.pi*c_CGS)
+    else:
+        gamma_stark = 0
```

Comment:
Here is what you could do:
```python
# determine temperature coefficient. Adiabatic theory summarized in Rivière 2002 - Eq. (24-29) - doi:10.1016/S0022-4073(01)00173-X
if "is_ionized":
   n_Stark = -0.5 #according to adiabatic theory, dipole and quadripole moment scale with <1/v> = T**-0.5
else: #for neutrals
   n_Stark = 1/6 #according to adiabatic theory, dipole moment is constant and quadripole moment scale with <v>**(1/3) = T**(1/6)
gamma_stark = (10**gamSta) * (T/10000)**n_Stark * P*1e6*diluent['e-'] / (k_b_CGS*T) / (4*np.pi*c_CGS)
```

Reply:
>    n_Stark = 1/6 #according to adiabatic theory, dipole moment is constant and quadripole moment scale with <v>**(1/3) = T**(1/6)
How are the dipole and quadrupole gammas combined? If they're added together, then wouldn't this line be equivalent of scaling both the dipolar and quadrupolar gamma by `T**(1/6)` when the former is supposed to be constant?

Reply:
Yes you're right. There is no good answer. Let's keep the dependence in T**[1/6] to make the comparison with other codes easier